src/lib/useContractAddress.ts

```diff
@@ -1,13 +1,22 @@
-import { useParams } from "react-router-dom";
+import { useNavigate, useParams } from "react-router-dom";
 import { Address } from "ton";
+import { useEffect } from "react";
 
 function useContractAddress() {
+  const navigate = useNavigate();
   const { contractAddress } = useParams();
   let isAddressValid = validateAddress(contractAddress);
 
+  let verifiedAddress = isAddressValid ? Address.parse(contractAddress ?? "").toFriendly() : null;
```

Comment:
Address.parse on "" would throw.
```suggestion
  let verifiedAddress = isAddressValid ? Address.parse(contractAddress!).toFriendly() : null;
```